src/librustc/ty/constness.rs

```diff
@@ -68,14 +69,57 @@ impl<'tcx> TyCtxt<'tcx> {
 
 
 pub fn provide(providers: &mut Providers<'_>) {
-    /// only checks whether the function has a `const` modifier
+    fn is_const_intrinsic(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
+        // Const evaluability whitelist is here to check evaluability at the
+        // top level beforehand.
+        match tcx.fn_sig(def_id).abi() {
+            Abi::RustIntrinsic |
+            Abi::PlatformIntrinsic => {
+                match &tcx.item_name(def_id).as_str()[..] {
+                    | "size_of"
+                    | "min_align_of"
+                    | "needs_drop"
+                    | "type_id"
+                    | "bswap"
+                    | "bitreverse"
+                    | "ctpop"
+                    | "cttz"
+                    | "cttz_nonzero"
+                    | "ctlz"
+                    | "ctlz_nonzero"
+                    | "overflowing_add"
+                    | "overflowing_sub"
+                    | "overflowing_mul"
+                    | "unchecked_shl"
+                    | "unchecked_shr"
+                    | "rotate_left"
+                    | "rotate_right"
+                    | "add_with_overflow"
```

Comment:
The wrapping intrinsics seem to be missing from this list. The errors mean the check is working! Awesome! Please add anything to this list that is needed to get the tests passing